library/GitHubRelease.hs

```diff
@@ -75,7 +98,22 @@ upload aToken anOwner aRepo aTag aFile aName = do
   response <- uploadFile manager uploadUrl aToken aFile aName
   case HTTP.statusCode (Client.responseStatus response) of
     201 -> pure ()
-    _ -> fail "Failed to upload file to release!"
+    _   -> fail "Failed to upload file to release!"
+
+release :: String -> Maybe String -> String -> String -> String -> Maybe String -> Maybe Bool -> Maybe Bool -> IO ()
+release aToken anOwner aRepo aTag aTitle aDescription aPreRelease aDraft = do
+  manager <- Client.newManager TLS.tlsManagerSettings
+  (owner', repo') <- getOwnerRepo anOwner aRepo
+  let format = "https://api.github.com/repos/%s/%s/releases"
+  let
+    url :: String
+    url = Printf.printf format owner' repo'
+  response <- mkRelease manager url aToken aTag aTitle aDescription aPreRelease aDraft
+  let body = Aeson.eitherDecode $ Client.responseBody response :: Either String Aeson.Object
+  case HTTP.statusCode (Client.responseStatus response) of
+    201 -> pure ()
+    422 -> IO.hPutStrLn IO.stderr "Release aready exists. Ignoring."
```

Comment:
One more note: I'm not entirely sure this is 100% correct or if 422 can be some other sort of error. One option would be to decode the json error message and look for "code": "already_exists" to be sure.

Reply:
Then again the go tool does the same: `error: github returned 422 Unprocessable Entity (this is probably because the release already exists)`